dali/operators/reader/loader/webdataset/tar_utils.cc

```diff
@@ -202,6 +215,23 @@ inline bool TarArchive::ParseHeader() {
   } else {
     filename_ = th_get_pathname(ToTarHandle(handle_));
     filesize_ = th_get_size(ToTarHandle(handle_));
+
+    if (TH_ISREG(ToTarHandle(handle_))) {
+      filetype_ = ENTRY_FILE;
+    } else if (TH_ISDIR(ToTarHandle(handle_))) {
+      filetype_ = ENTRY_DIR;
+    } else if (TH_ISLNK(ToTarHandle(handle_))) {
+      filetype_ = ENTRY_HARDLINK;
+    } else if (TH_ISSYM(ToTarHandle(handle_))) {
+      filetype_ = ENTRY_SYMLINK;
+    } else if (TH_ISCHR(ToTarHandle(handle_))) {
+      filetype_ = ENTRY_CHARDEV;
+    } else if (TH_ISBLK(ToTarHandle(handle_))) {
+      filetype_ = ENTRY_BLOCKDEV;
+    } else {
+      DALI_ENFORCE(TH_ISFIFO(ToTarHandle(handle_)), "Entry type in a tar archive not recognized");
+      filetype_ = ENTRY_FIFO;
+    }
```

Comment:
Just an idea. This is OK. But I think it'd be more readable as:
```
} else if (TH_ISFIFO(tar_handle))) {
  file_type = ENTRY_FIFO;
} else {
  DALI_FAIL("Entry type not recognized"); 
}
```

Reply:
Alternatively, you can use `switch` ;)

Reply:
alright; also, regarding switch, how would I go about it?

Reply:
I don't think you can use switch with those TH_ISFIFO and such